src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -4121,18 +4122,15 @@ class StubGenerator: public StubCodeGenerator {
     Register c         = x14; // a4
     Register d         = x15; // a5
 
-    Register state0    =  x5; // t0
-    Register state1    =  x6; // t1
-    Register state2    =  x7; // t2
-    Register state3    =  x9; // s1
+    Register state0    =  x7; // t2
+    Register state1    = x20; // s4
+    Register state2    = x21; // s5
+    Register state3    = x22; // s6
```

Comment:
Seems to me the comment before the head of this method is kind of duplicated with this part in the code.
I would recommend to merge the 2 part here, and remove the comment before the head of this method (LINE 4048-4080). By doing this, 
1. the reader is not distracted to read 2 parts of similar code to get the information;
2. when modify the code, writer can just modify one place.

But thsi is quite subjective, and it's piece of existing code, so it depends on you.

Reply:
I like having the table in the comment before the code to have a high level view of which registers are used for what. This is less obvious in the code. But that's my person opinion!